rs/recovery/src/app_subnet_recovery.rs

```diff
@@ -132,9 +133,11 @@ pub struct AppSubnetRecoveryArgs {
     #[clap(long)]
     pub keep_downloaded_state: Option<bool>,
 
-    /// IP address of the node to upload the new subnet state to
-    #[clap(long)]
-    pub upload_node: Option<IpAddr>,
+    /// The method of uploading state. Possible values are either `local` (for a
+    /// local recovery on the admin node) or the ipv6 address of the target node.
+    /// Local recoveries allow us to skip a potentially expensive data transfer.
+    #[clap(long, value_parser=crate::upload_method_from_str)]
+    pub upload_method: Option<UploadMethod>,
```

Comment:
Nice! Did you consider doing something similar for the download node?

Reply:
Would a "local download" be different from simply skipping the download step?

Reply:
Yes, we would simply create a cheap copy of the state on top of which we run ic-replay

Reply:
Ah right, let me do that in a subsequent PR then